jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -169,6 +172,25 @@ def blink_once(self):
 
                 for ilayer in (set(valid) - set([next_layer])):
                     self.state.layers[ilayer].visible = False
+                self.set_compass(self.state.layers[next_layer].layer)
+
+    def set_compass(self, image):
+        """Update the Compass plugin with info from the given image Data object."""
+        if self.compass is None:
+            try:
+                self.compass = Compass(app=self)
```

Comment:
This hack didn't work. I'll have to refactor to either implement a new message Compass can subscribe to, or wait for https://jira.stsci.edu/browse/JDAT-2015 .

Either way, I am not liking much the compass not connected to the image like this:

![Screenshot 2021-12-09 224147](https://user-images.githubusercontent.com/2090236/145515691-d8ba2d63-8e8b-426d-8885-32357984e3dc.jpg)

People are used to the compass drawn on a smaller version of the image, like this:

![DS9](https://astrobites.org/wp-content/uploads/2011/03/ds91.png)

So, I am afraid that implementing this plugin will be a waste of time if eventually we have to re-implement it in a different way.